File: js/algo/2stack.js
class twoStack
{
    constructor()
    {
        this.stack1 = [];
        this.stack2 = [];
    }

    push1(item)
    {
        this.stack1.push(item);
    }

    push2(item)
    {
        this.stack2.push(item);
    }

    pop1()
    {
        if(this.stack1.length == 0)
        {
            console.log("Stack Underflow");
            return;
        }
        return this.stack1.pop();
    }

    pop2()
    {
        if(this.stack2.length == 0)
        {
            console.log("Stack Underflow");
            return;
        }
        return this.stack2.pop();
    }

    peek1()
    {
        return this.stack1[this.stack1.length - 1];
    }

    peek2()
    {
        return this.stack2[this.stack2.length - 1];
    }

    isEmpty1()
    {
        return this.stack1.length == 0;
    }

    isEmpty2()
    {
        return this.stack2.length == 0;
    }

    change1(element, i)
    {
        this.stack1[i] = element;
    }

    change2(element, i)
    {
        this.stack2[i] = element;
    }

    printStack1()
    {
        let str = "";
        for(let i = 0; i < this.stack1.length; i++)
        {
            str += this.stack1[i] + " ";
        }
        return str;
    }

    printStack2()
    {
        let str = "";
        for(let i = 0; i < this.stack2.length; i++)
        {
            str += this.stack2[i] + " ";
        }
        return str;
    }
}

let stack = new twoStack();
stack.push1(21);
stack.push1(22);
stack.push1(23);
stack.push1(24);
stack.push1(25);

stack.push2(31);
stack.push2(32);
stack.push2(33);
stack.push2(34);
stack.push2(35);

console.log(stack.printStack1());
console.log(stack.printStack2());

File: cpp/algo/2stack.cpp
#include <bits/stdc++.h>
using namespace std;
#define FAST_IO ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0); cerr.tie(0)
typedef long long ll;
typedef unsigned long long ul;
typedef long double ld;
#define mp make_pair
#define gl getline
#define pb push_back
#define eb emplace_back
#define ob pop_back
#define sz(s) ((int)(s.size()))
#define UM uno\nrdered_map
#define US uno\nrdered_set
#define forn(i, n) for (int i = 1; i <= int(n); i++)
#define fora(i, n) for(auto i:n)
#define Len 1005
const double pi=3.14159265358979323846;
const int MOD = 1000000007;

class 2_Stack 
{
    int top;
    int 
    public:
        ll a[Len]; 
        
        bool push1(ll x);
        ll pop1();
        ll peek1();
        bool isEmpty1();
        bool isFull1();
        void change1(int idx, ll element);
        void display1();

        bool push2(ll x);
        ll pop2();
        ll peek2();
        bool isEmpty2();
        bool isFull2();
        void change2(int idx, ll element);
        void display2();
};

  
// Driver program to test above functions
int main()
{
    class 2_Stack s;
    s.push(21);
    s.push(22);
    s.push(24);
    s.push(25);
    s.display(); cout << endl;
    cout << s.pop() << endl;
    cout << s.pop() << endl;
    s.display();
}